docs/contributing.md

```diff
@@ -12,13 +12,14 @@ If you're a first-time contributor, you can check the issues with [good first is
 2. Clone the repository from your GitHub.
 3. Setup development environment through [poetry](https://python-poetry.org/) (`poetry install`).
 4. Setup [pre-commit](https://pre-commit.com/) hook (`pre-commit install -t pre-commit -t pre-push -t commit-msg`)
-5. Check out a new branch and add your modification.
-6. Add test cases for all your changes.
+5. Install [gpg](https://gnupg.org) to enable tests
```

Comment:
I might suggest extracting it as a standalone section on this page. We can call it a `prerequisite`. `poetry` is one of such `prerequiste`